salt/modules/cmdmod.py

```diff
@@ -415,7 +415,8 @@ def _get_stripped(cmd):
         if isinstance(cmd, (list, tuple)):
             cmd = ' '.join(map(_cmd_quote, cmd))
 
-        cmd = 'su -l {0} -c "cd {1}; {2}"'.format(runas, cwd, cmd)
+        cmd = 'cd {0} && {1}'.format(_cmd_quote(cwd), cmd)
```

Comment:
What us line 411 trying to do. Seems there is two sections for Darwin.  Is this code it the right spot (not your fault)
If you look at this function eventual for all platforms except for Windows, this is turned into a list.
cmd = [ 'su', '-l', runas, '-c', 'cd \'{0}\' && {1}'.format(_cmd_quote(cwd), cmd) ]

Reply:
I understand your proposal and it seems reasonable to me, but I am not sure if `cmd` will be properly quoted, since it could contain single quotes that potentially can break the `-c` part of the command.

By doing this way:

```python
cmd = 'cd {0} && {1}'.format(_cmd_quote(cwd), cmd)
cmd = 'su -l {0} -c {1}'.format(runas, _cmd_quote(cmd))
```

`_cmd_quote` should ensure that, whatever the `cmd` content is, it will be properly escaped. Because previously `cmd` has been converted into a string with all its components escaped.